src/random.cpp

```diff
@@ -380,61 +362,128 @@ void LockingCallbackOpenSSL(int mode, int i, const char* file, int line) NO_THRE
     }
 }
 
-static void AddDataToRng(void* data, size_t len, RNGState& rng);
+static void SeedTimestamp(CSHA512& hasher)
+{
+    int64_t perfcounter = GetPerformanceCounter();
+    hasher.Write((const unsigned char*)&perfcounter, sizeof(perfcounter));
+}
 
-void RandAddSeedSleep()
+static void SeedFast(CSHA512& hasher)
 {
-    RNGState& rng = GetRNGState();
+    unsigned char buffer[32];
 
-    int64_t nPerfCounter1 = GetPerformanceCounter();
-    std::this_thread::sleep_for(std::chrono::milliseconds(1));
-    int64_t nPerfCounter2 = GetPerformanceCounter();
+    // Stack pointer to indirectly commit to thread/callstack
+    const unsigned char* ptr = buffer;
+    hasher.Write((const unsigned char*)&ptr, sizeof(ptr));
 
-    // Combine with and update state
-    AddDataToRng(&nPerfCounter1, sizeof(nPerfCounter1), rng);
-    AddDataToRng(&nPerfCounter2, sizeof(nPerfCounter2), rng);
+    // Hardware randomness is very fast when available; use it always.
```

Comment:
Assuming it's always CPU based and not same external device, so this assumption should be stated in the empty / placeholder `HWRandInit` implementations for other architectures.

Reply:
Done.